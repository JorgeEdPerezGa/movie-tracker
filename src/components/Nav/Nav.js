import React from 'react';
import { NavLink } from 'react-router-dom';
import './Nav.css';

const Nav = () => {
  return(
    <nav className='nav-menu'>
      <NavLink className='nav' to='/movies'>Movies</NavLink>
      <NavLink className='nav' to='/login'>Login</NavLink>
      <NavLink className='nav' to='/register'>Register</NavLink>
      <NavLink className='nav' to='/favorites'>Favorites</NavLink>
    </nav>
  )
}

export default Nav;