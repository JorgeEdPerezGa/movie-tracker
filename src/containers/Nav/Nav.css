.nav-menu {
  background-color: purple;
  padding: 20px; }

.nav {
  color: white;
  text-decoration-line: none; }

.active {
  color: lightblue; }
