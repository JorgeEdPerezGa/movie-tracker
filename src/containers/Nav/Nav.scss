.nav-menu {
  background-color: #393839;
  display: flex;
  padding: 20px;
  justify-content: space-between;
}

.nav {
  color: white;
  text-decoration-line: none;
}

.menu {
  width: 60%;
  font-size: 17px;
  display: flex;
  justify-content: space-around;
}

.active {
  color: lightblue;
}

.app-title-1 {
  font-size: 20px;
  margin: 0;
  color: white;
  font-weight: 400;
}

.app-title-2 {
 color: #83cdff;
}

.home-nav {
  text-decoration: none;
  color: white;
}
