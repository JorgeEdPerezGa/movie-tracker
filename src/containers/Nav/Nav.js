import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { initialFetch, cleanMovies } from '../../helper';
import { addMovies, logoutUser } from '../../actions';
import './Nav.css';

export class Nav extends Component {

  async componentDidMount(){
    const response = await initialFetch();
    const movies = await cleanMovies(response);
    this.props.addMovies(movies);
  }

  handleLogoutUser = () => {
    const { user, logoutUser } = this.props;

    logoutUser(user);
  }

  render() {
    if (this.props.user.name) {
      return (
        <nav className='nav-menu'>
          <p
            className='app-title-1'>
          <NavLink to='/' className='home-nav'>
            <span
              className='app-title-2'>MOVIE
            </span>TRACKER
          </NavLink>
          </p>
          <div className='menu'>
            <NavLink className='nav' to='/movies'>Movies</NavLink>
            <NavLink className='nav' to='/favorites'>Favorites</NavLink>
            <button onClick={this.handleLogoutUser}>Logout</button>
          </div>
        </nav>
      );
    }

    return (
      <nav className='nav-menu'>
        <p
          className='app-title-1'>
          <NavLink to='/' className='home-nav'>
            <span
              className='app-title-2'>MOVIE
            </span>TRACKER
          </NavLink>
        </p>
        <div className='menu'>
          <NavLink className='nav' to='/movies'>Movies</NavLink>
          <NavLink className='nav' to='/login'>Login</NavLink>
          <NavLink className='nav' to='/register'>Register</NavLink>
        </div>
      </nav>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  logoutUser: (user) => dispatch(logoutUser(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
