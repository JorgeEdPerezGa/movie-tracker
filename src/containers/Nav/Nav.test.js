import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';

describe('Nav', () => {
  let renderedComponent;
  let mockAddMovies;
  let mockLogoutUser;
  let mockClearFavorites;
  let mockUser;
  let mockMoviesData;
  let mockHistory;

  beforeAll(() => {
    mockMoviesData = {
      results: [
        {
          adult: false,
          backdrop_path: "/dddddd.jpg",
          genre_ids: [12, 16, 35, 10751],
          id: 354912,
          original_language: "en",
          original_title: "Coco",
          overview: "Despite his famil...",
          popularity: 207.580955,
          poster_path: "/ffff.jpg",
          release_date: "2017-10-27",
          title: "Coco",
          video: false,
          vote_average: 7.7
        },
        {
          adult: false,
          backdrop_path: "/zxc.jpg",
          genre_ids: [123, 10751],
          id: 99999,
          original_language: "sp",
          original_title: "Nono",
          overview: "Hellooooo there...",
          popularity: 207.580955,
          poster_path: "/phd.jpg",
          release_date: "2011-11-21",
          title: "Nono",
          video: false,
          vote_average: 1.7
        }
      ]
    };

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve(mockMoviesData)
    }));
  });

  beforeEach(() => {
    mockAddMovies = jest.fn();
    mockLogoutUser = jest.fn();
    mockClearFavorites = jest.fn();
    mockHistory = { push: jest.fn(), go: jest.fn() };
    mockUser = {
      email: "fake@123.com",
      id: 2,
      name: "ted",
      password: "pw123"
    };
    renderedComponent = shallow(
      <Nav
        user={mockUser}
        addMovies={mockAddMovies}
        logoutUser={mockLogoutUser}
        clearFavorites={mockClearFavorites}
        history={mockHistory}
      />
    );
  });

  it('should match snapshot when no user is logged in', () => {
    mockUser = {};
    renderedComponent = shallow(
      <Nav
        user={mockUser}
        addMovies={mockAddMovies}
        logoutUser={mockLogoutUser}
        clearFavorites={mockClearFavorites}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match snapshot when user is logged in', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should logout user when user clicks logout button and logoutUser is called', () => {
    renderedComponent.instance().handleLogoutUser();
    expect(mockLogoutUser).toHaveBeenCalledWith(mockUser);
  });

  it('should clear favorites in store when user clicks logout button and clearFavorites is called', () => {
    renderedComponent.instance().handleLogoutUser();
    expect(mockClearFavorites).toHaveBeenCalled();
  });
});
