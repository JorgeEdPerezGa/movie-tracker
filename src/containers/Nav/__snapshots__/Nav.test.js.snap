// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot when no user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    addMovies={[Function]}
    logoutUser={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="app-title-1"
>
          <span
                    className="app-title-2"
          >
                    MOVIE
          </span>
          TRACKER
</p>,
        <div
          className="menu"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/movies"
          >
                    Movies
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/login"
          >
                    Login
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/register"
          >
                    Register
          </NavLink>
</div>,
      ],
      "className": "nav-menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="app-title-2"
>
              MOVIE
</span>,
            "TRACKER",
          ],
          "className": "app-title-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MOVIE",
              "className": "app-title-2",
            },
            "ref": null,
            "rendered": "MOVIE",
            "type": "span",
          },
          "TRACKER",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/movies"
>
              Movies
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/login"
>
              Login
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/register"
>
              Register
</NavLink>,
          ],
          "className": "menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Movies",
              "className": "nav",
              "to": "/movies",
            },
            "ref": null,
            "rendered": "Movies",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Login",
              "className": "nav",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Register",
              "className": "nav",
              "to": "/register",
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="app-title-1"
>
            <span
                        className="app-title-2"
            >
                        MOVIE
            </span>
            TRACKER
</p>,
          <div
            className="menu"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/movies"
            >
                        Movies
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/login"
            >
                        Login
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/register"
            >
                        Register
            </NavLink>
</div>,
        ],
        "className": "nav-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="app-title-2"
>
                MOVIE
</span>,
              "TRACKER",
            ],
            "className": "app-title-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MOVIE",
                "className": "app-title-2",
              },
              "ref": null,
              "rendered": "MOVIE",
              "type": "span",
            },
            "TRACKER",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/movies"
>
                Movies
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/login"
>
                Login
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/register"
>
                Register
</NavLink>,
            ],
            "className": "menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Movies",
                "className": "nav",
                "to": "/movies",
              },
              "ref": null,
              "rendered": "Movies",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login",
                "className": "nav",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Register",
                "className": "nav",
                "to": "/register",
              },
              "ref": null,
              "rendered": "Register",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Nav should match snapshot when user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    addMovies={[Function]}
    logoutUser={[Function]}
    user={
        Object {
            "email": "fake@123.com",
            "id": 2,
            "name": "ted",
            "password": "pw123",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="app-title-1"
>
          <span
                    className="app-title-2"
          >
                    MOVIE
          </span>
          TRACKER
</p>,
        <div
          className="menu"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/movies"
          >
                    Movies
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/favorites"
          >
                    Favorites
          </NavLink>
          <button
                    onClick={[Function]}
          >
                    Logout
          </button>
</div>,
      ],
      "className": "nav-menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="app-title-2"
>
              MOVIE
</span>,
            "TRACKER",
          ],
          "className": "app-title-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MOVIE",
              "className": "app-title-2",
            },
            "ref": null,
            "rendered": "MOVIE",
            "type": "span",
          },
          "TRACKER",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/movies"
>
              Movies
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/favorites"
>
              Favorites
</NavLink>,
            <button
              onClick={[Function]}
>
              Logout
</button>,
          ],
          "className": "menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Movies",
              "className": "nav",
              "to": "/movies",
            },
            "ref": null,
            "rendered": "Movies",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Favorites",
              "className": "nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="app-title-1"
>
            <span
                        className="app-title-2"
            >
                        MOVIE
            </span>
            TRACKER
</p>,
          <div
            className="menu"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/movies"
            >
                        Movies
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/favorites"
            >
                        Favorites
            </NavLink>
            <button
                        onClick={[Function]}
            >
                        Logout
            </button>
</div>,
        ],
        "className": "nav-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="app-title-2"
>
                MOVIE
</span>,
              "TRACKER",
            ],
            "className": "app-title-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MOVIE",
                "className": "app-title-2",
              },
              "ref": null,
              "rendered": "MOVIE",
              "type": "span",
            },
            "TRACKER",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/movies"
>
                Movies
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/favorites"
>
                Favorites
</NavLink>,
              <button
                onClick={[Function]}
>
                Logout
</button>,
            ],
            "className": "menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Movies",
                "className": "nav",
                "to": "/movies",
              },
              "ref": null,
              "rendered": "Movies",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Favorites",
                "className": "nav",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
